doc/reference/pipeline_spec.md

```diff
@@ -280,64 +341,9 @@ The datums are defined as whichever files or directories match by the glob patte
 
 ## Multiple Inputs
 
-A pipeline is allowed to have multiple inputs.  The important thing to
-understand is what happens when a new commit comes into one of the input repos.
-
-In short, a pipeline processes the **cross product** of the datums in its
-inputs.  We will
-use an example to illustrate.
-
-Consider a pipeline that has two input repos: `foo` and `bar`.
-
-```
-1. PUT /file-1 in commit1 in foo -- no jobs triggered
-2. PUT /file-a in commit1 in bar -- triggers job1
-3. PUT /file-2 in commit2 in foo -- triggers job2
-4. PUT /file-b in commit2 in bar -- triggers job3
-```
-
-The first time the pipeline is triggered will be when the second event
-completes.  This is because we need data in both repos before we can run the
-pipeline.
-
-Here is a breakdown of the datums that each job sees:
-
-```
-job1:
-
-Datum1:
-
-    /pfs/foo/file-1
-    /pfs/bar/file-a
-
-job2:
-
-Datum1:
-    /pfs/foo/file-1
-    /pfs/bar/file-a
-
-Datum2:
-    /pfs/foo/file-2
-    /pfs/bar/file-a
-
-job3:
-
-Datum1:
-    /pfs/foo/file-1
-    /pfs/bar/file-a
-
-Datum2:
-    /pfs/foo/file-2
-    /pfs/bar/file-a
-
-Datum3:
-    /pfs/foo/file-1
-    /pfs/bar/file-b
-
-Datum4:
-    /pfs/foo/file-2
-    /pfs/bar/file-b
-```
+It's important to note that if a pipeline takes multiple atom inputs (via cross
+or union) then the pipeline will not get triggered until all of the atom inputs
```

Comment:
Curious why it doesn't make sense for union inputs to be triggered before all atoms inputs have a commit? Couldn't the pipeline just run as soon as any datums are present?